astropy/modeling/fitting.py

```diff
@@ -1210,21 +1210,24 @@ def _wrap_deriv(params, model, weights, x, y, z=None):
             if z is None:
                 fit_deriv = np.array(model.fit_deriv(x, *params))
                 try:
+                    # Both np.array functions can raise a ValueError here and should be kept in the same try block
                     output = np.array(
                         [np.ravel(_) for _ in np.array(weights) * fit_deriv]
                     )
                     if output.shape != fit_deriv.shape:
                         output = np.array(
                             [np.ravel(_) for _ in np.atleast_2d(weights).T * fit_deriv]
                         )
-                    return output
                 except ValueError:
                     return np.array(
                         [
                             np.ravel(_)
                             for _ in np.array(weights) * np.moveaxis(fit_deriv, -1, 0)
                         ]
                     ).transpose()
+
+                else:
```

Comment:
I just don't see why this is any better.

Reply:
The try-except statement has an else clause for code that should
run only if no exceptions were raised. Using the else clause is more
explicit than using a return statement inside of a try block.

Reply:
I think the functionality is clear enough, the question here as with the `except: pass` cases above is probably rather how explicit the code has to be made, when all possible outcomes of this code branch are already present within 3 or 4 code lines – here it is either the `return output` when the `try` succeeds, the alternative `return` from `ValueError`, or raising an uncaught exception.
Though one might discuss here if it is even clearer to assign to `output` within the `except ValueError` as well and then always `return output` _after_ the whole `try/except` block.

Reply:
> Though one might discuss here if it is even clearer to assign to output within the except ValueError as well and then always return output after the whole try/except block.
I agree with this. Will soon open up a seperate PR for modelling.